### PR TITLE
feat: build with rollup

### PR DESCRIPTION
**Pull request by masone**
References https://autoricardo.atlassian.net/browse/CAR-9284

## Motivation and context

Moving away from pika/pack, following up on the RFC in https://github.com/carforyou/carforyou-docs/discussions/147 - this POC creates an esm and commonjs build with rollup. similar result to what we had before, just without the bloat of pika/pack

Thanks for the constant support @gzimh

### CODE REVIEW

File: jest.config.cjs [hunk 1]
```diff
@@ -9,8 +9,8 @@ module.exports = {
     "<rootDir>/__tests__/factories/",
     "<rootDir>/pkg/",
   ],
-  setupFiles: ["<rootDir>/__tests__/setup/fetchMock.ts"],
-  setupFilesAfterEnv: ["<rootDir>/__tests__/setup/setupConfiguration.ts"],
+  setupFiles: ["<rootDir>/src/__tests__/setup/fetchMock.ts"],
+  setupFilesAfterEnv: ["<rootDir>/src/__tests__/setup/apiClient.ts"],
```
Comment (masone) [inline jest.config.cjs:13]: moved to be included with a simpler tsconfig

File: rollup.config.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+export default [
+  {
+    input: "src/index.ts",
+    output: [
+      {
+        file: packageJson.main,
+        format: "cjs",
+        sourcemap: true,
+        name: "react-lib",
```
Comment (Averethel) [inline rollup.config.js:15]: Is this a leftover?

Comment (masone) [reply rollup.config.js:15]: thanks. yes (copy pastaing tutorials)

File: package.json [hunk 3]
```diff
@@ -22,33 +22,26 @@
...
-      [
-        "@pika/plugin-build-node"
-      ]
-    ]
+  "main": "pkg/cjs/index.js",
+  "module": "pkg/esm/index.js",
+  "types": "pkg/index.d.ts",
+  "files": [
+    "pkg/**/*"
+  ],
```
Comment (masone) [inline package.json:43]: pika was setting these on the generated package.json, they now are manually added to the root

File: package.json [hunk 4]
```diff
@@ -1,10 +1,10 @@
 {
   "name": "@carforyou/api-client",
-  "version": "9.1.0",
+  "version": "0.0.0",
   "description": "Api client to connect with our APIs",
   "scripts": {
     "version": "npm run build",
-    "build": "pika build",
+    "build": "rimraf pkg && rollup -c",
```
Comment (lkappeler) [inline package.json:7]: can't you just use `rm` directly in an npm command? (I have in mind it's just invoking the shell)

Comment (masone) [reply package.json:7]: good question. rimraf has cross-OS support, whereas `rm` is a linux-specific command. I also remember it having some safety measures such as not deleting files outside of your root directory (ergo no "oops, I committed a `rm -rf /`" and now everyone is fucked) and dealing nicely with directories not existing in the first place

**Comment by masone** [comment]
we decided to prefer a build with tsc for node packages

**Pull request closed without merging**